pkg/integrations/manager.go

```diff
@@ -116,6 +109,35 @@ func (c *ManagerConfig) DefaultRelabelConfigs(hostname string) []*relabel.Config
 	return cfgs
 }
 
+// ApplyDefaults applies default settings to the ManagerConfig and validates
+// that it can be used.
+//
+// If any integrations are enabled and are configured to be scraped, the
+// Prometheus configuration must have a WAL directory configured.
+func (c *ManagerConfig) ApplyDefaults(cfg *prom.Config) error {
+	var willCreateInstance bool
+
+	for _, ic := range c.Integrations {
+		if !ic.CommonConfig().Enabled {
+			continue
+		}
+
+		scrapeIntegration := c.ScrapeIntegrations
+		if common := ic.CommonConfig(); common.ScrapeIntegration != nil {
+			scrapeIntegration = *common.ScrapeIntegration
+		}
+		if scrapeIntegration {
+			willCreateInstance = true
```

Comment:
Could you return early here instead of using the willCreateInstance?

Reply:
ah, good point!